attestation-service/verifier/src/cca/mod.rs

```diff
@@ -60,6 +61,44 @@ struct CcaEvidence {
     token: Vec<u8>,
 }
 
+#[derive(Error, Debug)]
+pub enum CCAError {
+    #[error("CCA verifier must provide report data field!")]
+    ReportDataMissing,
+    #[error("Deserialize CCA Evidence failed: {0}")]
+    FailedtoDeserializeCCAEvidence(String),
+    #[error("CCA Attestation failed with error: {0}")]
+    CCAAttestationFailed(String),
+    #[error("get the decoding key from the pem public key: {0}")]
+    GetDecodingKey(String),
+    #[error("decrypt the ear with the decoding key: {0}")]
+    DecryptDecodingKey(String),
+    #[error("decode nonce byte from ear: {0}")]
+    DecodeNonceByte(String),
+    #[error("report data is different from that in ear's session nonce")]
+    ReportDataMismatch,
+    #[error("no entry found for CCA_SSD_PLATFORM")]
+    NoEntryFound,
+    #[error("init data hash is different from that in CCA token")]
+    InitDataHashMismatch,
+    #[error("Failed to decode base64: {0}")]
+    FailedtoDecodeBase64(String),
+    #[error("get platform evidence from the cca evidence map")]
+    GetPlatformEvidence,
+    #[error("Serde Json Error")]
+    SerdeJson(#[from] serde_json::Error),
+    #[error("get realm evidence from the cca evidence map")]
+    GetRealmEvidence,
+    #[error("CAA Verifier error: {0}")]
+    CAAVerifier(String),
+    #[error("verification error")]
+    Verification(#[from] veraison_apiclient::Error),
+    #[error("Failed to discover the verification endpoint details")]
+    VerificationEndpoint(),
+    #[error("anyhow error: {0}")]
+    Anyhow(#[from] anyhow::Error),
```

Comment:
I am worried about this kind of error. Other types have determined semantics while this not. Could you try to classify the source of this error as one of the above and use `map_err`?